zfit/core/parameter.py

```diff
@@ -465,23 +465,51 @@ class ComposedParameter(BaseComposedParameter):
     # TODO: raise error if eager is on (because it's very errorprone)
     def __init__(self, name, tensor, **kwargs):
         tensor = ztf.convert_to_tensor(tensor)
-        independend_params = tf.get_collection("zfit_independent")
-        params = get_dependents(tensor=tensor, candidates=independend_params)
+        independent_params = tf.get_collection("zfit_independent")
+        params = get_dependents(tensor=tensor, candidates=independent_params)
         # params_init_op = [param.initializer for param in params]
         params = {p.name: p for p in params}
         # with tf.control_dependencies(params_init_op):
         super().__init__(params=params, initial_value=tensor, name=name, **kwargs)
 
 
 class ComplexParameter(BaseComposedParameter):
```

Comment:
Actually, why not just inherit from ComposedParameter? It *is* a composed parameter and reduces code dublication

Reply:
Oh, so true! Fixed!

Reply:
Since `convert_to_tensor` has a hardcoded `dtype.float`, I updated `ComposedParameter`.